### PR TITLE
remove doctesting from refguide-check, add `spin check-tutorials`

### PR DESCRIPTION
**Pull request by ev-br**
Following up on https://github.com/numpy/numpy/pull/26604#issuecomment-2163776646,

- add a brief mention of `spin check-docs` in the devguide
- add `spin check-tutorials` command to doctest RST files
- remove doctesting from `refguide-check`

### CODE REVIEW

**Comment by ev-br** [comment]
Relevant CI is green: circleCI runs the non-doctest part of refguide-check, check-{docs,tutorials} are in the benchmarks linux run.

File: doc/source/dev/development_environment.rst [hunk 1]
```diff
@@ -114,6 +114,15 @@ argument to pytest::
 
     $ spin test -v -t numpy/_core/tests/test_multiarray.py -- -k "MatMul and not vector"
 
+To run "doctests" -- to check that the code examples in the documentation is correct --
+use the `check-docs` spin command. It relies on the `scipy-docs` package, which
+provides several additional features on top of the standard library ``doctest``
+package. Install ``scipy-doctest`` and run on of::
```
Comment (mattip) [inline doc/source/dev/development_environment.rst:120]:
```suggestion
package. Install ``scipy-doctest`` and run one of::
```

**Review by mattip — APPROVED** [review]
LGTM, except for one typo

**Comment by mattip** [comment]
Thanks @ev-br

**Pull request merged**